From 43ec43c0d889ecabcb06fbc6b334e884ad44a48f Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Wed, 15 Nov 2023 13:19:16 -0500
Subject: [PATCH 59/68] v4l2codecs: h264enc: Don't call virtual negotiate()
 directly

This is not meant to be used like this, call
gst_video_encoder_negotiate() instead.
Upstream-Status: Pending
---
 sys/v4l2codecs/gstv4l2codech264enc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/v4l2codecs/gstv4l2codech264enc.c b/sys/v4l2codecs/gstv4l2codech264enc.c
index 4591af3..e7ae63a 100644
--- a/sys/v4l2codecs/gstv4l2codech264enc.c
+++ b/sys/v4l2codecs/gstv4l2codech264enc.c
@@ -623,7 +623,7 @@ gst_v4l2_codec_h264_enc_set_format (GstVideoEncoder * encoder,
       gst_video_encoder_set_output_state (GST_VIDEO_ENCODER (self),
       caps, state);
 
-  if (GST_VIDEO_ENCODER_CLASS (parent_class)->negotiate (encoder)) {
+  if (gst_video_encoder_negotiate (encoder)) {
     if (!gst_v4l2_encoder_streamon (self->encoder, GST_PAD_SINK)) {
       GST_ELEMENT_ERROR (self, RESOURCE, FAILED,
           ("Could not enable the encoder driver."),
-- 
2.25.1

